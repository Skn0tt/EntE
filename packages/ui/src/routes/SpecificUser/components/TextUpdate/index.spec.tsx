import * as React from "react";
import { shallow } from "enzyme";
import * as sinon from "sinon";
import { TextUpdate } from ".";

describe("TextUpdate", () => {
  const onChange = sinon.spy();
  const comp = shallow(
    <TextUpdate
      value="Hello world"
      onChange={onChange}
      validator={_ => true}
      title="TextUpdate"
    />
  );

  it("renders correctly", () => {
    expect(comp).toMatchSnapshot();
  });

  describe("when typing", () => {
    comp.find(".updateInput").simulate("change", { target: { value: "test" } });

    it("fires `onChange`", () => {
      expect(onChange.calledWith("test")).toBe(true);
    });
  });
});
