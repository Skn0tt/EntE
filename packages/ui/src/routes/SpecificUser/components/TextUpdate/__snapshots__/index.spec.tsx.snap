// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextUpdate renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    onChange={[Function]}
    title="TextUpdate"
    validator={[Function]}
    value="Hello world"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithStyles(GridWrapper)
          item={true}
        >
          <WithStyles(Typography)
            variant="title"
          >
            TextUpdate
          </WithStyles(Typography)>
        </WithStyles(GridWrapper)>,
        <WithStyles(GridWrapper)
          item={true}
        >
          <TextField
            className="updateInput"
            error={false}
            fullWidth={true}
            onChange={[Function]}
            required={false}
            select={false}
            type={undefined}
            value="Hello world"
          />
        </WithStyles(GridWrapper)>,
      ],
      "container": true,
      "direction": "column",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <WithStyles(Typography)
            variant="title"
          >
            TextUpdate
          </WithStyles(Typography)>,
          "item": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "TextUpdate",
            "variant": "title",
          },
          "ref": null,
          "rendered": "TextUpdate",
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <TextField
            className="updateInput"
            error={false}
            fullWidth={true}
            onChange={[Function]}
            required={false}
            select={false}
            type={undefined}
            value="Hello world"
          />,
          "item": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "updateInput",
            "error": false,
            "fullWidth": true,
            "onChange": [Function],
            "required": false,
            "select": false,
            "type": undefined,
            "value": "Hello world",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(GridWrapper)
            item={true}
          >
            <WithStyles(Typography)
              variant="title"
            >
              TextUpdate
            </WithStyles(Typography)>
          </WithStyles(GridWrapper)>,
          <WithStyles(GridWrapper)
            item={true}
          >
            <TextField
              className="updateInput"
              error={false}
              fullWidth={true}
              onChange={[Function]}
              required={false}
              select={false}
              type={undefined}
              value="Hello world"
            />
          </WithStyles(GridWrapper)>,
        ],
        "container": true,
        "direction": "column",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(Typography)
              variant="title"
            >
              TextUpdate
            </WithStyles(Typography)>,
            "item": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "TextUpdate",
              "variant": "title",
            },
            "ref": null,
            "rendered": "TextUpdate",
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <TextField
              className="updateInput"
              error={false}
              fullWidth={true}
              onChange={[Function]}
              required={false}
              select={false}
              type={undefined}
              value="Hello world"
            />,
            "item": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "updateInput",
              "error": false,
              "fullWidth": true,
              "onChange": [Function],
              "required": false,
              "select": false,
              "type": undefined,
              "value": "Hello world",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
